account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -432,6 +432,23 @@ const generateSignedPermit = async <
     ],
   });
 
+  const [decimals_response, nonce_response, paymasterData] = await Promise.all([
```

Comment:
❔ (noChange) So why choose this style.
Since the futures are already in flight, is there a performance reason for this.

Wouldn't 
```ts
const decimals_response = await decimals_future;
const none_response = await nonce_future;
const paymasterData_future = await paymasterData_future;
```
Also why are these snake_cased and camelCased?

Reply:
An advantage of awaiting in parallel is that if any of the promises fail, then we can fail immediately. In your snippet, if `paymasterData_future` fails, then this function would still wait for `decimals_future` and `nonce_future` before terminating.

Agreed about the naming, we should use camelCase in TypeScript.